test/Microsoft.Azure.SignalR.AspNet.Tests/ServiceConnectionTests.cs

```diff
@@ -46,16 +48,20 @@ public async Task ServiceConnectionDispatchTest()
                     // Application layer sends OpenConnectionMessage
                     var openConnectionMessage = new OpenConnectionMessage(clientConnection, new Claim[0], null, "?transport=webSockets");
                     await proxy.WriteMessageAsync(new OpenConnectionMessage(clientConnection, new Claim[0]));
-
                     await proxy.WaitForClientConnectAsync(clientConnection).OrTimeout();
-
-                    // TODO: Check response when integrated with ServiceMessageBus
-                    await proxy.WriteMessageAsync(new ConnectionDataMessage(clientConnection, GetPayload("Hello World")));
-
-                    await proxy.WaitForApplicationMessageAsync(clientConnection).OrTimeout();
+                    
+                    while(count < 1000)
+                    {
+                        await proxy.WriteMessageAsync(new ConnectionDataMessage(clientConnection, GetPayload("Hello World")));
+                        await proxy.WaitForApplicationMessageAsync(clientConnection).OrTimeout();
+                        count++;
+                    }
+                    // Wait 1 second for async channel processing messages.
+                    await Task.Delay(TimeSpan.FromMilliseconds(1000));
```

Comment:
>1000 [](start = 63, length = 4)

might cause random test failure

Reply:
Add a tcs into the test transport and `awit tcs` instead

---
In reply to: [239367027](https://github.com/Azure/azure-signalr/pull/268#discussion_r239367027) [](ancestors = 239367027)

Reply:
Or send a close connection message

---
In reply to: [239367182](https://github.com/Azure/azure-signalr/pull/268#discussion_r239367182) [](ancestors = 239367182,239367027)

Reply:
Close is smart. Will move validation in the end.